creator-node/src/premiumContent/PremiumContentQueue.ts

```diff
@@ -0,0 +1,127 @@
+import { Queue, Worker } from 'bullmq'
+import config from '../config'
+import { logger } from '../logging'
+import { updatePremiumContentCIDCache } from './helpers'
+
+const models = require('../models')
+const { QueryTypes } = require('sequelize')
+
+const QUEUE_INTERVAL_MS = 6 * 60 * 60 * 1000 // 6 hours
+const PREMIUM_CONTENT_QUEUE_HISTORY = 500
+const QUEUE_NAME = 'premium-content-queue'
+const PROCESS_NAME = 'premium-content'
+
+/**
+ * A persistent cron-style queue that periodically refreshes
+ * the premium content CIDs and their corresponding track ids.
+ */
+export class PremiumContentQueue {
+  queue: Queue
+
+  constructor() {
+    const connection = {
+      host: config.get('redisHost'),
+      port: config.get('redisPort')
+    }
+    this.queue = new Queue(QUEUE_NAME, {
+      connection,
+      defaultJobOptions: {
+        removeOnComplete: PREMIUM_CONTENT_QUEUE_HISTORY,
+        removeOnFail: PREMIUM_CONTENT_QUEUE_HISTORY
+      }
+    })
+    this.queue.drain(true)
+    ;(() =>
+      new Worker(
+        QUEUE_NAME,
+        async (_) => {
+          await this.logStatus('Starting')
+          try {
+            const premiumContentCIDMap = await this.getPremiumContentCIDMap()
+            await updatePremiumContentCIDCache({
+              cacheMap: premiumContentCIDMap,
+              logger
+            })
+          } catch (e) {
+            this.logStatus(
+              `Error in premium content worker: ${(e as Error).message}`
+            )
+          }
+        },
+        { connection }
+      ))()
+  }
+
+  /**
+   * Joins the "Tracks" and "Files" tables on track ids and sees if the
+   * corresponding track is premium.
+   * Returns map of these premium CIDs to their corresponding track ids.
+   */
+  async getPremiumContentCIDMap() {
```

Comment:
this one is tricky because I see how we need all records to check for uniqueness, but we'd need to batch these to avoid loading all entries from Tracks+Files table into memory. I did some napkin math on the size of those tables for batching params [here](https://github.com/AudiusProject/audius-protocol/pull/3954#issuecomment-1267141518) and have a loosely related problem that I solved with Redis sets in that PR - pagination [here](https://github.com/AudiusProject/audius-protocol/pull/3954/files#diff-95ca4010be0ddb97115e70e5ee46e212370c284eaf5ecd6cd179b28cf68f8b14R274-R288) and query [here](https://github.com/AudiusProject/audius-protocol/pull/3954/files#diff-6074c2bd95d6310e102a64f8c40169dc90ccd7a86f66a1f95ab35e2ad1d72b32R178-R201)

Reply:
very good point. however, just updated the code so that only the premium cids will be loaded. can you please take another look at the query and logic and lmk if you think we still need the pagination? thx

Reply:
@sddioulde this seems sensible, have we tried running this query on a snapshotted prod data to get a sense of perf?

Reply:
I would want to err on the side of supporting higher amounts of premium content in the future with pagination but don't think it needs to block on this now if you and Piazza are happy with the query performance on prod. I am concerned with the massive # of rows in the Files table tho so curious to hear the runtime of this currently